include/swift/AST/SILOptions.h

```diff
@@ -121,7 +122,9 @@ class SILOptions {
   bool AssumeUnqualifiedOwnershipWhenParsing = false;
 
   /// Assume that code will be executed in a single-threaded environment.
-  bool AssumeSingleThreaded = false;
+  /// TODO: return to false when we pass "-Xfrontend -assume-single-threaded"
+  ///       to swift compiler while building standard library.
+  bool AssumeSingleThreaded = std::getenv("SWIFT_ASSUME_SINGLE_THREADED");
```

Comment:
It is a bit unorthodox to initialize the value of a compiler option from the environment variable. Once we have a better integration with the build system, we should better pass proper -assume-single-threaded options to the swift compiler rather than doing this trick.